### PR TITLE
Fix `--version` flag for cedana

### PR DESCRIPTION
**Pull request by yashanand1910**
## Describe your changes
It makes sense to embed the version from `git describe --tags` whenever building cedana. This way whenever we manually tag on GitHub, the release CI would automatically get that.

There were some simplifications to the CI workflows and the Dockerfile as a result, so on next release we should we keep an eye for any errors.

## Issue ticket number 
CED-578 & CED-586

## Checklist before requesting a review
- [x] Have I performed a self-review?
- [x] Have I added regression or unit tests (where it makes sense) for my changes?
- [x] Have I updated the README if changes have resulted in it being out of date?
- [x] Have I checked to ensure my PR only introduces the changes it's intended to? E.g. No extraneous formatting changes.
- [ ] Is this a breaking change? If yes, please describe areas affected and request reviews from developer stakeholders.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -51,6 +51,8 @@ jobs:
         with:
           fetch-depth: 0
           submodules: recursive
+          fetch-tags: true
```
Comment (swarnimarun) [inline .github/workflows/release.yml:54]: I believe fetch-depth 0 should fetch the entire history. Also note, fetch-depth 0 fetch the origin main head, which is fine for release but I am not sure if it's the correct behaviour we want.

**Review by swarnimarun — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -84,6 +86,7 @@ jobs:
         uses: actions/checkout@v4
         with:
           submodules: recursive
+          fetch-tags: true
```
[the remark this answers is not in the document]
Comment (yashanand1910) [reply .github/workflows/release.yml:89]: Good callout, as I'm reading online this should work when the action triggered by a tag. Still, will test it out!

File: .github/workflows/release.yml (hunk 1 above)
Comment (yashanand1910) [reply .github/workflows/release.yml:54]: Yeah, I wasn't sure why this was `0` in the first place, so I left it. It should be `1` here too. @nravic Was there a reason for this being `0`?

Comment (nravic) [reply .github/workflows/release.yml:54]: Hmm no reason

File: .github/workflows/release.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (yashanand1910) [reply .github/workflows/release.yml:89]: @swarnimarun Added a regression test to see if `cedana --version` works correctly. I think this should be sufficient to see if the `fetch-tags` is working correcly or now.

**Review by nravic — APPROVED** [review]
looks good!

**Pull request merged**